src/pt/sakuramangas/src/eu/kanade/tachiyomi/extension/pt/sakuramangas/SakuraMangas.kt

```diff
@@ -0,0 +1,331 @@
+            headers,
+            form.build(),
+        )
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val document = response.asJsoup()
+
+        val chapterId = document.selectFirst("meta[chapter-id]")!!.attr("chapter-id")
+        val token = document.selectFirst("meta[token]")!!.attr("token")
+
+        val response = client.newCall(pageListApiRequest(chapterId, token)).execute()
+            .parseAs<SakuraMangaChapterReadDto>()
+
+        val baseUrl = document.baseUri().trimEnd('/')
+
+        return response.imageUrls.mapIndexed { index, url ->
+            Page(
+                index,
+                imageUrl = "$baseUrl/$url".toHttpUrl().toString(),
+            )
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    override fun getFilterList(): FilterList {
+        CoroutineScope(Dispatchers.IO).launch {
+            fetchFilters()
+        }
+
+        return FilterList(
+            OrderByFilter("Ordenar por", orderByOptions, "order"),
+            DemographyFilter("Demografia", demographyOptions, "demography"),
+            ClassificationFilter("Classificação", classificationOptions, "classification"),
+            GenreList(
+                title = "Gêneros",
+                genres = genresSet.toTypedArray(),
+            ),
+        )
+    }
+
+    private fun fetchFilters() {
+        try {
+            val document = client
+                .newCall(GET("$baseUrl/obras/", headers))
+                .execute()
+                .asJsoup()
+
+            genresSet = document.select(".genero-badge").map { element ->
+                val id = element.attr("data-value")
+                Genre(element.ownText(), id)
+            }.toSet()
+
+            val demoOpts = document.select("select#demografia-select option").mapNotNull { opt ->
+                val value = opt.attr("value").orEmpty()
+                val text = opt.text().trim()
+                if (text.isEmpty()) null else text to value
+            }
+            if (demoOpts.isNotEmpty()) demographyOptions = demoOpts
+
+            val classOpts =
+                document.select("select#classificacao-select option").mapNotNull { opt ->
+                    val value = opt.attr("value").orEmpty()
+                    val text = opt.text().trim()
+                    if (text.isEmpty()) null else text to value
+                }
+            if (classOpts.isNotEmpty()) classificationOptions = classOpts
+
+            val orderOptions = document.select("select#ordenar-por option").mapNotNull { opt ->
+                val value = opt.attr("value").orEmpty()
+                val text = opt.text().trim()
+                if (text.isEmpty()) null else text to value
+            }
+            if (orderOptions.isNotEmpty()) orderByOptions = orderOptions
+        } catch (e: Exception) {
+            Log.e(tag, e.stackTraceToString())
```

Comment:
```suggestion
            Log.e("SakuraMangas", "failed to fetch genres", e)
```

And remove the `tag`.